internal/operators/handler/handler.go

```diff
@@ -0,0 +1,134 @@
+func (h *Handler) ListSupportedOperators(_ context.Context, _ restoperators.ListSupportedOperatorsParams) middleware.Responder {
+	return restoperators.NewListSupportedOperatorsOK().
+		WithPayload(h.operatorsAPI.GetSupportedOperators())
+}
+
+// ListOfClusterOperators Lists operators to be monitored for a cluster.
+func (h *Handler) ListOfClusterOperators(ctx context.Context, params restoperators.ListOfClusterOperatorsParams) middleware.Responder {
+	operatorsList, err := h.GetMonitoredOperators(ctx, params.ClusterID, params.OperatorName, h.db)
+	if err != nil {
+		return common.GenerateErrorResponder(err)
+	}
+	return restoperators.NewListOfClusterOperatorsOK().WithPayload(operatorsList)
+}
+
+// ReportMonitoredOperatorStatus Controller API to report of monitored operators.
+func (h *Handler) ReportMonitoredOperatorStatus(ctx context.Context, params restoperators.ReportMonitoredOperatorStatusParams) middleware.Responder {
+	err := h.UpdateMonitoredOperatorStatus(ctx, params.ClusterID, params.ReportParams.Name, params.ReportParams.Status, params.ReportParams.StatusInfo)
+	if err != nil {
+		return common.GenerateErrorResponder(err)
+	}
+	return restoperators.NewReportMonitoredOperatorStatusOK()
+}
+
+// GetMonitoredOperators retrieves list of monitored operators for a cluster
+func (h *Handler) GetMonitoredOperators(ctx context.Context, clusterID strfmt.UUID, operatorName *string, db *gorm.DB) (models.MonitoredOperatorsList, error) {
+	log := logutil.FromContext(ctx, h.log)
+	var where *gorm.DB
+	if operatorName == nil || *operatorName == "" {
+		where = db.Where("cluster_id = ?", clusterID)
+	} else {
+		where = db.Where("name = ? and cluster_id = ?", *operatorName, clusterID)
+	}
+
+	var operatorsList = models.MonitoredOperatorsList{}
+	if err := where.Find(&operatorsList).Error; err != nil {
+		log.WithError(err).Errorf("failed to find monitored operators")
+		if gorm.IsRecordNotFoundError(err) {
+			return nil, common.NewApiError(http.StatusNotFound, err)
+		}
+	}
+
+	return operatorsList, nil
+}
+
+// UpdateMonitoredOperatorStatus updates status and status info of a monitored operator for a cluster
+func (h *Handler) UpdateMonitoredOperatorStatus(ctx context.Context, clusterID strfmt.UUID, monitoredOperatorName string, status models.OperatorStatus, statusInfo string) error {
+	log := logutil.FromContext(ctx, h.log)
+	if monitoredOperatorName == "" {
+		return common.NewApiError(http.StatusBadRequest, errors.New("empty operator name is not allowed"))
+	}
+	txSuccess := false
+	tx := h.db.Begin()
+	defer func() {
+		if !txSuccess {
+			log.Error("update monitored operator failed")
+			tx.Rollback()
+		}
+		if r := recover(); r != nil {
+			log.Error("update monitored operator failed")
+			tx.Rollback()
+		}
+	}()
+
+	monitoredOperators, err := h.GetMonitoredOperators(ctx, clusterID, &monitoredOperatorName, tx)
+	if err != nil {
+		return err
+	}
+	if len(monitoredOperators) == 0 {
```

Comment:
Please add a check for a number of monitored operators greater than 1 (shouldn't happen. but let's log and fail in case this happens)

Reply:
I didn't put that check here because the operator name is part of a primary key and the DB shouldn't allow for creation of two operators with the same name for one cluster. 
What value do you see in that check? I see only one situation it can fail: when the DB schema is broken and cluster is created with more than one operator with the same name.

Reply:
You are right. But let's assume the scheme might change in the future. I think that adding an extra verification in the function level is harmless. :)
Otherwise, maybe having another wrapping function for getting a singular `MonitoredOperator` (by name) instead of an array would be useful and much clearer.

Reply:
Went with option no. 2 :)